src/services/cache/RedisCache.ts

```diff
@@ -170,43 +183,58 @@ export class RedisCache {
 
     // Redis requires that subscribers use a separate client
     this.subscriberClient = this.client.duplicate();
```

Comment:
This seems that it is only necessary for the Redis client, not the Redis cluster.

https://github.com/redis/node-redis/blob/master/docs/pub-sub.md
![image](https://user-images.githubusercontent.com/779802/235915995-8c72fd04-4476-47e3-9dfa-3eec0bb419b0.png)

Reply:
I've switched the GB Proxy over to `ioredis` since `node-redis` has some issues with flakiness and incomplete TS bindings (specifically for cluster mode). Ioredis recommends using a separate client for subscribing even in cluster mode. While the general Redis docs indicate that you don't technically need to separate your subscribe client from normal usage due to how the Cluster message bus works, it's still a recommended practice to separate them (scalability, separation of concerns, flexibility) and there's no real downside to doing so.

Of course this is just my own research and I have not battle-tested this at scale. If you have scalability insights here, would be great to learn of them.